src/main/java/org/eclipse/uprotocol/cloudevent/datamodel/UCloudEventAttributes.java

```diff
@@ -40,28 +40,32 @@ public class UCloudEventAttributes {
     private final UPriority priority;
     private final Integer ttl;
     private final String token;
+    private final String traceparent;
 
     /**
      * Construct the properties object.
      *
-     * @param hash     an HMAC generated on the data portion of the CloudEvent message using the device key.
+     * @param hash     An HMAC generated on the data portion of the CloudEvent message using the device key.
      * @param priority uProtocol Prioritization classifications.
      * @param ttl      How long this event should live for after it was generated (in milliseconds).
      *                 Events without this attribute (or value is 0) MUST NOT timeout.
      * @param token    Oauth2 access token to perform the access request defined in the request message.
+     * @param traceparent    Optional identifier used to correlate observability across related events
      */
-    private UCloudEventAttributes(String hash, UPriority priority, Integer ttl, String token) {
+    private UCloudEventAttributes(String hash, UPriority priority, Integer ttl, String token, String traceparent) {
```

Comment:
Optional attributes are not passed to the constructor, I fully expect traceparent to be an optional attribute

Reply:
Thanks for the feedback @stevenhartley .  I overloaded the constructor and it is now optional.  Does this resolve your comment, or are you suggesting that optional parameters may only be set through the builder pattern.

Reply:
That is not what I meant :-). constructor is for mandatory parameters, don't touch the constructor and add a builder method for this parameter.

Reply:
Done